packages/amplify-e2e-tests/src/__tests__/graphql-v2/searchable-node-to-node-encryption/searchable-previous-deployment-had-node-to-node.test.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  initJSProjectWithProfile,
+  deleteProject,
+  amplifyPush,
+  createRandomName,
+  addAuthWithDefault,
+  setTransformConfigValue,
+  removeTransformConfigValue,
+} from 'amplify-category-api-e2e-core';
+import { addApiWithoutSchema, updateApiSchema, getProjectMeta } from 'amplify-category-api-e2e-core';
+import { createNewProjectDir, deleteProjectDir } from 'amplify-category-api-e2e-core';
+import gql from 'graphql-tag';
+import AWSAppSyncClient, { AUTH_TYPE } from 'aws-appsync';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+
+(global as any).fetch = require('node-fetch');
+
+describe('searchable deployment when previous deployed state had node to node encryption enabled', () => {
+  let projRoot: string;
+  let projectName: string;
+  let appSyncClient = undefined;
+
+  beforeEach(async () => {
+    projectName = createRandomName();
+    projRoot = await createNewProjectDir(createRandomName());
+    await initJSProjectWithProfile(projRoot, {
+      name: projectName,
+    });
+    await addAuthWithDefault(projRoot, {});
+  });
+
+  afterEach(async () => {
+    await deleteProject(projRoot);
+    deleteProjectDir(projRoot);
+  });
+
+  it('if previous deployment had NodeToNodeEncryption, carry it forward, there should be no data loss', async () => {
+    const v2Schema = 'transformer_migration/searchable-v2.graphql';
+
+    await addApiWithoutSchema(projRoot, { apiName: projectName });
+    updateApiSchema(projRoot, projectName, v2Schema);
+
+    const searchableStackPath = path.join(projRoot, 'amplify', 'backend', 'api', projectName, 'build', 'stacks', 'SearchableStack.json');
+
+    // Initial Deploy with flag explicitly set
+    setTransformConfigValue(projRoot, projectName, 'NodeToNodeEncryption', true);
+    await amplifyPush(projRoot);
+
+    appSyncClient = getAppSyncClientFromProj(projRoot);
+    await runAndValidateQuery('test1', 'test1', 10, 1); // Expect a single record
+
+    const searchableStackFirstDeploy = JSON.parse(fs.readFileSync(searchableStackPath).toString());
+    const searchDomainPropsFirstDeploy = searchableStackFirstDeploy.Resources.OpenSearchDomain.Properties;
+
+    expect(searchDomainPropsFirstDeploy).toHaveProperty('NodeToNodeEncryptionOptions');
+    expect(searchDomainPropsFirstDeploy.NodeToNodeEncryptionOptions.Enabled).toEqual(true);
+
+    // Subsequent deploy without flag set
+    removeTransformConfigValue(projRoot, projectName, 'NodeToNodeEncryption');
+    await amplifyPush(projRoot);
+
+    appSyncClient = getAppSyncClientFromProj(projRoot);
+    await runAndValidateQuery('test1', 'test1', 10, 2); // Expect two records
```

Comment:
In this case, shouldn't the search domain be re-created? Since, now we're turning the encryption off.  How does getting back 2 records test that, sorry if I missed something here.

Reply:
Behavior of the systems is such that we will maintain the current state of nodeToNode (either on or off), so there should be no loss of data. The explicit flag is being removed to test the state that "previous deployment had it, I don't care how, now I'm doing a normal deploy"